### DISCUSSION TITLE
In v6, method convertToDatabaseValueSQL() from custom type is not invoked properly when calling it within the nested embeddables.

### DISCUSSION THREAD

**Discussion by sevenryze** [mikro-orm/mikro-orm#5074] [General]
### Describe the bug

Method convertToDatabaseValueSQL() from custom type is not invoked properly when calling it within the nested embeddables. It gets error: DriverException: Trying to serialize raw SQL fragment.

### Reproduction

```typescript
import {
  Embeddable,
  Embedded,
  Entity,
  MikroORM,
  PrimaryKey,
  Property,
  Type,
} from '@mikro-orm/sqlite';

class GeoPoint {

  constructor(
    readonly latitude: number,
    readonly longitude: number,
  ) {}

}

class GeoPointType extends Type<GeoPoint | undefined, string | undefined> {

  override convertToDatabaseValue(
    value: GeoPoint | undefined,
  ): string | undefined {
    if (!value) {
      return value;
    }

    return `point(${value.latitude} ${value.longitude})`;
  }

  override convertToJSValue(value: string | undefined): GeoPoint | undefined {
    const m = value?.match(/point\((-?\d+(\.\d+)?) (-?\d+(\.\d+)?)\)/i);

    if (!m) {
      return undefined;
    }

    return new GeoPoint(+m[1], +m[3]);
  }

  override convertToJSValueSQL(key: string) {
    return `ST_AsText(${key})`;
  }

  override convertToDatabaseValueSQL(key: string) {
    return `ST_PointFromText(${key}, 4326)`;
  }

  override getColumnType(): string {
    return 'point SRID 4326';
  }

}

@Embeddable()
class SnapshotMore {

  @Property({
    type: GeoPointType,
  })
  position!: GeoPoint;

  @Property({
    nullable: true,
  })
  description?: string;

}

@Embeddable()
class Snapshot {

  @Embedded(() => SnapshotMore, {
    object: true,
  })
  ref!: SnapshotMore;

}

@Entity()
class Outer {

  @PrimaryKey()
  id!: number;

  @Embedded(() => Snapshot, {
    object: true,
  })
  bigSnapshot!: Snapshot;

}

describe('GH issue 1003', () => {

  let orm: MikroORM;

  beforeAll(async () => {
    orm = await MikroORM.init({
      dbName: ':memory:',
      debug: true,
      entities: [Outer, Snapshot],
    });
    await orm.schema.createSchema();
  });

  afterAll(async () => {
    await orm.close(true);
  });

  test(`GH issue 1003`, async () => {
    const em = orm.em.fork();

    const snapshotMore = new SnapshotMore();
    snapshotMore.position = new GeoPoint(21, 32);

    const snapshot = new Snapshot();
    snapshot.ref = snapshotMore;

    const outer = new Outer();
    outer.id = 123_456;
    outer.bigSnapshot = snapshot;

    await em.persistAndFlush([outer]);

    const result = await em.findOne(Outer, {
      id: outer.id,
    });
  });

});
```

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

6.0.0-rc3

### Node.js version

v20.3.1

### Operating system

MacOS 14.2.1

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
This is a wontfix, those SQL converters are only supported on actual database columns, while a property on nested object embedded entity is just a part of a JSON object.

I will look into why it fails this way, but the bottom line is - you are trying something that's not supposed to be working (the SQL converter is not supposed to be fired at all for object embeddables).

**Reply by sevenryze** [reply]
Understood, it makes sense. Also, thank you for such a quick response!

**Reply by B4nan** [reply]
I think the proper fix to this will be to skip the convertors completely if we see we are inside object embedded property, that will fix it universally for all drivers.

**Reply by B4nan** [reply]
This should help 83b989ebdb2adc27c2f42f717f5d95abd01c109c
